File: core/src/main/java/com/example/coinstest/data/CurrencyInRubRepository.java
package com.example.coinstest.data;

import com.example.coinstest.domain.CurrencyInRub;

import java.time.LocalDate;

public class CurrencyInRubRepository {
    private CurrencyInRubDataSource currencyDataSource;

    public CurrencyInRubRepository(CurrencyInRubDataSource dataSource) {
        currencyDataSource = dataSource;
    }

    public CurrencyInRub getCurrencyOnDate(LocalDate date) {
        return currencyDataSource.getCurrencyOnDate(date);
    }

    public LocalDate getLastDate() {
        return currencyDataSource.getLastDate();
    }
}

File: core/src/main/java/com/andysklyarov/finnotifyfree/data/CurrencyInRubRepository.kt
package com.andysklyarov.finnotifyfree.data

import com.andysklyarov.finnotifyfree.domain.CurrencyInRub
import io.reactivex.Single
import java.time.LocalDate

class CurrencyInRubRepository(private val currencyDataSource: CurrencyInRubDataSource) {

    fun getCurrencyOnDate(currencyCode: String, date: LocalDate): Single<CurrencyInRub> {
        return currencyDataSource.getCurrencyOnDate(currencyCode, date)
    }

    fun getCurrencyOnPreviousDate(currencyCode: String, date: LocalDate): Single<CurrencyInRub> {
        return currencyDataSource.getCurrencyOnPreviousDate(currencyCode, date)
    }

    fun getLastDate(): Single<LocalDate>{
        return currencyDataSource.getLastDate()
    }
}